### PR TITLE
fix: check status when deleting

### LINKED ISSUE #16558: [UI - Dashboard] Delete dashboard does not trigger notification
__Steps to reproduce:__
List the minimal actions needed to reproduce the behavior.

1. Create a dashboard
2. From the dashboards page click the trash can icon
3. Confirm delete

__Expected behavior:__
Describe what you expected to happen.

Expected a success notification message, "Dashboard <NAME> deleted successfully"

__Actual behavior:__
Describe What actually happened.

There was no notification.   

Console contains: 

```
Uncaught (in promise) TypeError: Cannot read property 'message' of undefined
    at thunks.ts:251
```
__Screencast__

DashboardDeleteNoMessage01

__Environment info:__

* System info: Run `uname -srm` and copy the output here
`Linux 4.15.0-74-generic x86_64`
* InfluxDB version: Run `influxd version` and copy the output here
Detected in nightly build. 
```
ts=2020-01-16T08:56:11.465672Z lvl=info msg="Welcome to InfluxDB" log_id=0KNOvLnG000 version=nightly commit=d753aa445 build_date=2020-01-16T08:56:11Z
```
* Other relevant environment details: Container runtime, disk info, etc
Browser: __Chrome 79__

__Config:__
Copy any non-default config values here or attach the full config as a gist or file.

Run with end to end testing enabled

```
--e2e-testing=true
```

__Logs:__
Include snippet of errors in log.
```
Uncaught (in promise) TypeError: Cannot read property 'message' of undefined
    at thunks.ts:251
(anonymous) @ thunks.ts:251
async function (async)
(anonymous) @ thunks.ts:241
(anonymous) @ index.js:8
(anonymous) @ redux.js:449
I.handleClick @ ContextMenuItem.tsx:55
s @ react-dom.production.min.js:14
h @ react-dom.production.min.js:15
(anonymous) @ react-dom.production.min.js:15
x @ react-dom.production.min.js:17
T @ react-dom.production.min.js:18
M @ react-dom.production.min.js:18
E @ react-dom.production.min.js:21
Dn @ react-dom.production.min.js:84
As @ react-dom.production.min.js:250
je @ react-dom.production.min.js:39
Tn @ react-dom.production.min.js:86
(anonymous) @ react-dom.production.min.js:251
t.unstable_runWithPriority @ scheduler.production.min.js:18
Fs @ react-dom.production.min.js:251
Ln @ react-dom.production.min.js:85
D.Object.isExtensible.Object.isExtensible.e.___hb.e.___hb @ honeybadger.js:466
```

### PR DESCRIPTION
**Pull request by drdelambre**
Closes #16558

copy pasta malaise

### CODE REVIEW

File: ui/src/dashboards/actions/thunks.ts [hunk 1]
```diff
@@ -240,7 +240,7 @@ export const deleteDashboard = (dashboard: Dashboard) => async (
   try {
     const resp = await api.deleteDashboard({dashboardID: dashboard.id})
 
-    if (resp.data !== 204) {
+    if (resp.status !== 204) {
```
Comment (ebb-tide) [inline ui/src/dashboards/actions/thunks.ts]: why is the type for data in the generated routes set to `any`?
```
interface DeleteDashboardNoContentResult {
  status: 204
  headers: Headers
  data: any
}
```

Comment (drdelambre) [reply ui/src/dashboards/actions/thunks.ts]: @121watts got any context on this?

Comment (ebb-tide) [reply ui/src/dashboards/actions/thunks.ts]: .. the swagger is defined as.. 
```
      responses:
        '204':
          description: Delete has been accepted
```
and the response actually comes back as 
```
status: 204
headers: {}
data: undefined
```

Comment (ebb-tide) [reply ui/src/dashboards/actions/thunks.ts]: but it looks like oats just creates an `any` data. that's annoying.. but not sure where it can be fixed.

**Pull request merged**